File: RootTools2/src/main/java/com/rarnu/tools/neo/xposed/ads/FuckDownload.java
package com.rarnu.tools.neo.xposed.ads;

import com.rarnu.tools.neo.xposed.XpUtils;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

/**
 * Created by rarnu on 11/18/16.
 */
public class FuckDownload {

    public static void fuckDownload(XC_LoadPackage.LoadPackageParam loadPackageParam) {
        XpUtils.findAndHookMethod("com.android.providers.downloads.ui.recommend.config.ADConfig", loadPackageParam.classLoader, "OSSupportAD", new XC_MethodHook() {
            protected void beforeHookedMethod(MethodHookParam paramAnonymousMethodHookParam) throws Throwable {
                paramAnonymousMethodHookParam.setResult(false);
            }
        });
        XpUtils.findAndHookMethod("com.android.providers.downloads.ui.utils.BuildUtils", loadPackageParam.classLoader, "isCmTestBuilder", XC_MethodReplacement.returnConstant(true));
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/xposed/ads/FuckDownload.kt
package com.rarnu.tools.neo.xposed.ads

import com.rarnu.xfunc.*

/**
 * Created by rarnu on 11/18/16.
 */
object FuckDownload {

    fun fuckDownload(pkg: XposedPkg) {
        pkg.findClass("com.android.providers.downloads.ui.recommend.config.ADConfig").findMethod("OSSupportAD").hook { replace { result = false } }
        pkg.findClass("com.android.providers.downloads.ui.utils.CloudConfigHelper").apply {
            findMethod("supportRank").hook { replace { result = false } }
            findMethod("isShouldShowAd").hook { replace { result = false } }
            findMethod("isShouldShowRecommendInfo").hook { replace { result = false } }
            findMethod("getHomeAdRefreshType").hook { replace { result = 0 } }
            findMethod("isShouldShowExtraAd").hook { replace { result = false } }
            findMethod("isShouldShowRankGuide").hook { replace { result = false } }
        }
    }
}
